rskj-core/src/main/java/co/rsk/mine/gas/provider/MinGasPriceProviderFactory.java

```diff
@@ -0,0 +1,44 @@
+package co.rsk.mine.gas.provider;
+
+import co.rsk.config.mining.StableMinGasPriceSystemConfig;
+import co.rsk.mine.gas.provider.onchain.OnChainMinGasPriceProviderFactory;
+import co.rsk.mine.gas.provider.web.WebStableMinGasPriceProviderFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class MinGasPriceProviderFactory {
+
+    private static final Logger logger = LoggerFactory.getLogger("StableMinGasPrice");
+
+    private MinGasPriceProviderFactory() {
+    }
+
+    public static MinGasPriceProvider create(long fixedMinGasPrice, StableMinGasPriceSystemConfig stableMinGasPriceSystemConfig) {
+        FixedMinGasPriceProvider fixedMinGasPriceProvider = new FixedMinGasPriceProvider(fixedMinGasPrice);
+        if (stableMinGasPriceSystemConfig == null) {
+            logger.warn("Could not find stable min gas price system config, using {} provider", fixedMinGasPriceProvider.getType().name());
+            return fixedMinGasPriceProvider;
+        }
+        if (!stableMinGasPriceSystemConfig.isEnabled()) {
+            return fixedMinGasPriceProvider;
+        }
+        MinGasPriceProviderType method = stableMinGasPriceSystemConfig.getMethod();
+        if (method == null) {
+            logger.error("Could not find valid method in config, using fallback provider: {}", fixedMinGasPriceProvider.getType().name());
+            return fixedMinGasPriceProvider;
+        }
+
+
+        switch (method) {
+            case WEB:
+                return WebStableMinGasPriceProviderFactory.create(stableMinGasPriceSystemConfig, fixedMinGasPriceProvider);
+            case ON_CHAIN:
+                return OnChainMinGasPriceProviderFactory.create(stableMinGasPriceSystemConfig, fixedMinGasPriceProvider);
```

Comment:
suggestion: as commented previously, I would create the `MinGasPriceProvider` implementation here directly according the method. You already have all the configurations here, you could have methods to get the config according each type passed to extract the configuration complexity creation. 

The concept of a factory is exactly hold these logics necessary to return different implementations, if start to become too complex, we can abstract more and create other classes. But for now, the first level of abstraction is more than enough.